gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -188,6 +188,13 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     end
     T::Private::DeclState.current.reset!
 
+    if method_name == :method_added || method_name == :singleton_method_added
+      raise(
+        "Putting a `sig` on `#{method_name}` is not supported" +
+        " (sorbet-runtime uses this method internally to perform `sig` validation logic)"
+      )
+    end
```

Comment:
How does this interact with the `skip_on_method_added` above? Should this be before that?

Reply:
Do you have test cases for this error?

Reply:
We want this to be after the check for `current_declaration.nil?`, because we only want to raise in the case where someone tried to put a sig on these methods. We want to allow users to define the methods without siging them. (That's what I was thinking.)

This isn't tested but I'll write one.